### PR TITLE
adds cleanSpecialCharacters

### PR DESCRIPTION
**Pull request by Dbelataris**

### CODE REVIEW

File: converters/clean_special_characters.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package converters
```
Comment (JensSkipr) [inline converters/clean_special_characters.go:1]: As it's for strings, I would propose to add it to file `converters/strings.go` instead of creating a new file.

File: converters/clean_special_characters.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import (
+	"regexp"
+)
+
+/*
+CleanSpecialCharacters remove all special characters from a string input
+*/
+func CleanSpecialCharacters(input string) string {
+	regExp := regexp.MustCompile(`[^0-9a-zA-Z]+`)
```
Comment (JensSkipr) [inline converters/clean_special_characters.go:11]: Is it intentional you ditch letters with accents (e.g. `é`) as well? If not, please first pass the input through `converters.NormaliseString`.

File: converters/clean_special_characters.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package converters
+
+import (
+	"regexp"
+)
+
+/*
+CleanSpecialCharacters remove all special characters from a string input
```
Comment (JensSkipr) [inline converters/clean_special_characters.go:8]: Could you please be very explicit in the comments about what is considered a "special character"? I think it would be clearer to put something like `CleanSpecialCharacters removes any character which is not a digit or a character from the input`.

File: converters/clean_special_characters_test.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+func TestCleanSpecialCharacters(t *testing.T) {
+	var tests = []struct {
+		name   string
+		input  string
+		output string
+	}{
+		{"Remove special characters", "1111.222-B!?*5", "1111222B5"},
+		{"Remove spaces", "1111		AA 2", "1111AA2"},
```
Comment (JensSkipr) [inline converters/clean_special_characters_test.go:16]: Could you please add a test on an empty string and happy case (nothing removed) as well?

File: converters/clean_special_characters.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+*/
+func CleanSpecialCharacters(input string) string {
```
Comment (JensSkipr) [inline converters/clean_special_characters.go:10]: Could you please add this function to the `README`?

**Review by JensSkipr — CHANGES_REQUESTED** [review]

File: converters/clean_special_characters.go (hunk 2 above)
Comment (Dbelataris) [reply converters/clean_special_characters.go:11]: Very Good catch ! Thanks

File: converters/clean_special_characters.go (hunk 1 above)
Comment (Dbelataris) [reply converters/clean_special_characters.go:1]: no problem, I'll do it

File: converters/clean_special_characters.go (hunk 3 above)
Comment (Dbelataris) [reply converters/clean_special_characters.go:8]: yes indeed, your proposition is much better

File: converters/clean_special_characters.go (hunk 5 above)
Comment (Dbelataris) [reply converters/clean_special_characters.go]: yes sure

File: README.md [hunk 6]
```diff
@@ -56,6 +56,9 @@ normalised, genErr := converters.NormaliseString("Tëst Çôdé (test-result)")
 
 // Converts a string to snake_case
 output := converters.ToSnakeCase("ThisIS_a-veryRandom_string") // output == "this_is_a_very_random_string"
+
+// Removes any character(included spaces) which is not a digit or a letter from a string to snake_case
```
Comment (JensSkipr) [inline README.md:60]: Copy/paste issue `to snake_case`?

File: converters/strings_test.go [hunk 7]
```diff
@@ -49,3 +49,25 @@ func Test_ToSnakeCase(t *testing.T) {
...
+func Test_CleanSpecialCharacters(t *testing.T) {
+	var tests = []struct {
+		name   string
+		input  string
+		output string
+	}{
+		{"Input already clean", "1111AA2", "1111AA2"},
+		{"Remove special characters", "1111.222-B!?*5", "1111222B5"},
+		{"Remove spaces", "1111		AA 2", "1111AA2"},
+		{"Normalisse input", "1111AA 2Ç", "1111AA2C"},
```
Comment (JensSkipr) [inline converters/strings_test.go:62]: Typo `Normalisse`

**Review by JensSkipr — APPROVED** [review]
